src/compas/geometry/shapes/box.py

```diff
@@ -744,3 +744,52 @@ def contains_points(self, points, tol=1e-6):
             results[index] = True
 
         return results
+
+    def has_collision(self, other):
+        """Verify if box collides with another box.
+
+        Parameters
+        ----------
+        other : :class:`compas.geometry.Box`
+            The other box.
+
+        Returns
+        -------
+        bool
+            True if the two boxes collide.
+            False otherwise.
+
+        """
+        # get separation plane
+        def GetSeparatingPlane(RPos, axis, self, other):
+            return abs(RPos.dot(axis)) > (
+                abs((self.frame.xaxis * self.width * 0.5).dot(axis))
+                + abs((self.frame.yaxis * self.depth * 0.5).dot(axis))
+                + abs((self.frame.zaxis * self.height * 0.5).dot(axis))
+                + abs((other.frame.xaxis * other.width * 0.5).dot(axis))
+                + abs((other.frame.yaxis * other.depth * 0.5).dot(axis))
+                + abs((other.frame.zaxis * other.height * 0.5).dot(axis))
+            )
+
+        # compute the obb collision
+        RPos = other.frame.point - self.frame.point
```

Comment:
`RPos` is a local variable, should be lower-cased, with underscores if needed. it's also not clear what `rpos` stands for, i would go explicit, is it `relative_position`?